rust/nasl-interpreter/src/built_in_functions/cryptography/aes_ctr.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2023 Greenbone Networks GmbH
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+use crate::error::FunctionErrorKind::GeneralError;
+use aes::{
+    cipher::{
+        BlockCipher, BlockDecrypt, BlockEncrypt, BlockSizeUser, KeyIvInit, StreamCipher,
+        StreamCipherSeek,
+    },
+    Aes128, Aes192, Aes256,
+};
+use digest::typenum::U16;
+use sink::Sink;
+
+use crate::{error::FunctionError, NaslFunction, NaslValue, Register};
+
+use super::{get_named_data, get_named_number, Crypt};
+
+fn ctr<D>(register: &Register, crypt: Crypt, function: &str) -> Result<NaslValue, FunctionError>
```

Comment:
Currently panics if data is not exactly a multiple of block size (16 bytes). Padding is currently missing!

Reply:
Does not panic, as CTR ciphers byte streams, it is able to cipher a text of any length